beacon-chain/p2p/pubsub.go

```diff
@@ -147,9 +147,39 @@ func (s *Service) pubsubOptions() []pubsub.Option {
 		pubsub.WithGossipSubParams(pubsubGossipParam()),
 		pubsub.WithRawTracer(gossipTracer{host: s.host}),
 	}
+
+	if len(s.cfg.StaticPeers) > 0 {
+		s.addDirectPeersOption(&psOpts)
+	}
+
 	return psOpts
 }
 
+// addDirectPeersOption parses the static peers from the service configuration and adds
+// the direct peers option to the pubsub options.
+func (s *Service) addDirectPeersOption(psOpts *[]pubsub.Option) {
+	addrs, err := PeersFromStringAddrs(s.cfg.StaticPeers)
+
+	if err != nil {
+		log.WithError(err).Error("Could not convert static peers raw ENRs into multiaddresses")
+		return
+	}
+
+	if len(addrs) == 0 {
+		log.Warn("Converting static peers raw ENRs into multiaddresses resulted in an empty list. Ignoring direct peers configuration")
+		return
+	}
+
+	directAddrInfos, err := peer.AddrInfosFromP2pAddrs(addrs...)
+
+	if err != nil {
+		log.WithError(err).Error("Could not configure direct peers")
+		return
+	}
+
+	*psOpts = append(*psOpts, pubsub.WithDirectPeers(directAddrInfos))
```

Comment:
This isn't the standard way its usually done in go. To make this more readable can you return `psOpts` in the method ? so that it is clear what is the argument to the method and what is its output.

Reply:
I'm sorry again! I think I've simplified the logic a bit by moving the parsing of raw ENRs into a separate function. I hope it is idiomatic Go code now. Thanks for the patience

Reply:
No worries, it looks good to me now